File: problem/Reverse_Vowels/reverseVowels.java
class Solution {
    public String reverseVowels(String s) {
        char[] chars = s.toCharArray();
        int n = chars.length;
        int low = 0;
        int high = n - 1;
        String vowels = "aeiouAEIOU";

        while (low < high) {
            if (vowels.indexOf(chars[low]) != -1 && vowels.indexOf(chars[high]) != -1) {
                char temp = chars[low];
                chars[low] = chars[high];
                chars[high] = temp;
                low++;
                high--;
            } else {
                if (vowels.indexOf(chars[high]) == -1) {
                    high--;
                }
                if (vowels.indexOf(chars[low]) == -1) {
                    low++;
                }
            }
        }

        return new String(chars);
    }
}

File: problem/Reverse_Vowels/reverseVowels.py
class Solution:
    def reverseVowels(self, s):
        s = list(s)  
        n = len(s)
        low = 0
        high = n - 1
        vowels = set("aeiouAEIOU")
        
        while low < high:
            if s[low] in vowels and s[high] in vowels:
                s[low], s[high] = s[high], s[low]  
                low += 1
                high -= 1
            else:
                if s[high] not in vowels:
                    high -= 1
                if s[low] not in vowels:
                    low += 1
        
        return ''.join(s)
